backend/internal/service/app_config_service.go

```diff
@@ -362,6 +383,13 @@ func (s *AppConfigService) InitDbConfig() error {
 
 // LoadDbConfigFromDb loads the configuration values from the database into the DbConfig struct.
 func (s *AppConfigService) LoadDbConfigFromDb() error {
+	s.mu.Lock()         // Acquire write lock
+	defer s.mu.Unlock() // Release write lock
+	return s.loadDbConfigFromDbLocked()
+}
+
+// Add an unexported version that doesn't acquire the lock.
+func (s *AppConfigService) loadDbConfigFromDbLocked() error {
 	dbConfigReflectValue := reflect.ValueOf(s.DbConfig).Elem()
```

Comment:
In here you initialize a new *model.AppConfig (maybe that's a clone of the value you load from the pointer), work on that, then when you're done you call `s.DbConfig.Store()`